### PR TITLE
[WIP] Query GrahpQL GitHub Endpoint

### PR DESCRIPTION
**Pull request by kelset**
### DO NOT MERGE YET

I've decided to start reporting this PR earlier because it has many moving parts and I'd prefer to make sure I'm going in the right direction before going too far - it started with the discussion in #20 (that will get closed after this PR will get merged I guess).

#### Recap of things that happens

* a quick prettier + typos fix for all files in `./scripts`
* Added a new implementation of the `fetch-github-data.js` logic that uses  `apollo-fetch` to query the GraphQL GitHub endpoint
* Added a new field `lastMasterCommitData` that is basically a more reliable timestamp to evaluate when a lib has been updated (check issue for more details)
* To query the new GraphQL API an OAuth token is necessary, so I had to add to my local `secret.json` a new field `GITHUB_OATH_TOKEN` that I have generated via my profile.

#### Issues

* a couple fields, `hasPages` and `hasDownloads`, are not available via the GraphQL endpoint (or, at least, I couldn't find them) - but it also looks like they are not used (AFAIK)
* when a lib is not "available" or something (haven't investigated properly), the endpoint returns `null`, reason why I had to add a check in the `fetchGithubDataViaGraphQL` method and filter after `loadRepositoryDataAsync` in `build-and-score-data.js`
* There is no easy way to parse all topics per repo (AFAIK), need to set an upper bound number (now it's 10)

#### TODO

- [ ] fix topics parsing
- [ ] move fetch-license to graphql (or even in the same call as fetch-github-data)
- [ ] use the new `lastMasterCommitData` in the score calculation
- [ ] ??? (*suggestions*)

### CODE REVIEW

File: scripts/fetch-github-data.js [hunk 1]
```diff
@@ -1,9 +1,56 @@
 import 'isomorphic-fetch';
-import { GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET } from '../secrets.json';
+import { createApolloFetch } from 'apollo-fetch';
+
+import {
+  GITHUB_CLIENT_ID,
+  GITHUB_CLIENT_SECRET,
+  GITHUB_OATH_TOKEN,
```
Comment (jimmylee) [inline scripts/fetch-github-data.js]: OATH -> OAUTH?

Comment (kelset) [reply scripts/fetch-github-data.js]: whoops! Typo ;) Will fix it asap

**Comment by jimmylee** [comment]
:+1: Thanks for updating some of our other contributions with prettier.

So far it looks good, look forward to seeing you continue on this :)

**Comment by kelset** [comment]
Thanks for the feedback ;) I'll be on holiday starting Friday and I don't think I'll be able to move it forward before a couple weeks, but yeah once I'm back I hope I'll be able to complete it quickly.

**Comment by jimmylee** [comment]
@kelset this PR is a little stale at the moment, are you planning on continuing this work?

**Comment by kelset** [comment]
Hi @jimmylee - I'm so sorry I completely forgot about this PR 😓

Things happened. But tbh I still think I'd love to complete it - what about in a month? And if I can't manage to do it feel free to close it 😊

**Comment by jimmylee** [comment]
@kelset completely up to you!

**Comment by kelset** [comment]
(I'll close this cause I don't have time to work on it, I'm so sorry 😔)

**Pull request closed without merging**